programs/eosio-launcher/main.cpp

```diff
@@ -1271,6 +1304,12 @@ launcher_def::write_bios_boot () {
          string key = line.substr(len);
          if (key == "envars") {
             brb << "bioshost=" << bhost << "\nbiosport=" << biosport << "\n";
+            char cluster_str[20];
+            sprintf(cluster_str, "cluster%05d", cluster_id);
+            brb << "wddir=$wddir/" << (const char *)cluster_str << "\n";
+            brb << "wdaddr=localhost:" << calculate_port(cluster_id, 0, port_type::keosd_boot) << "\n";
```

Comment:
Why is a `keosd_boot` port type necessary that is separate from the other `keosd` port type?

Reply:
There's are 2 keosd processes for a cluster: the ignition wallet which is inside the bios_boot.sh, and the normal wallet which is managed in Cluster.py. I am thinking to make the ignition wallet to use port of ```calculate_port(cluster_id, 0, port_type::keosd)```, and the normal wallet use ```calculate_port(cluster_id, 1, port_type::keosd)```